### PR TITLE
census: add per call latency metric

### PR DESCRIPTION
**Pull request by DNVindhya**
This PR adds a new measure to calculate time taken by gRPC to complete an RPC (i.e `grpc.io/client/api_latency`).

It also enables corresponding view `GRPC_CLIENT_API_LATENCY_VIEW` for `grpc-gcp-observability`.

Note: New measure and view are defined in `grpc-census` temporarily. This will be moved to [opencensus-java](https://github.com/census-instrumentation/opencensus-java) repository later.

CC @ejona86 @sanjaypujare

### CODE REVIEW

**Comment by sanjaypujare** [comment]
> Note: New measure and view are defined in grpc-census temporarily. This will be moved to [opencensus-java](https://github.com/census-instrumentation/opencensus-java) repository later.

Since opencensus is deprecated and the repo most likely doesn't accept new PRs I am not sure if you will be able to do that. Just saying...

File: census/src/main/java/io/grpc/census/internal/ObservabilityCensusConstants.java [hunk 1]
```diff
@@ -37,6 +39,20 @@ public final class ObservabilityCensusConstants {
   static final Aggregation AGGREGATION_WITH_BYTES_HISTOGRAM =
       RpcViewConstants.GRPC_CLIENT_SENT_BYTES_PER_RPC_VIEW.getAggregation();
 
+  public static final MeasureDouble API_LATENCY_PER_CALL =
```
Comment (sanjaypujare) [inline census/src/main/java/io/grpc/census/internal/ObservabilityCensusConstants.java:42]: For line 36 can you add a TODO item to denote the code movement. I know it applies to the previous PR

File: census/src/main/java/io/grpc/census/internal/ObservabilityCensusConstants.java [hunk 2]
```diff
@@ -37,6 +39,20 @@ public final class ObservabilityCensusConstants {
...
+          "grpc.io/client/api_latency",
+          "Time taken by gRPC to complete an RPC from application's perspective",
+          "ms");
+
+  public static final View GRPC_CLIENT_API_LATENCY_VIEW =
+      View.create(
+          View.Name.create("grpc.io/client/api_latency"),
+          "Time taken by gRPC to complete an RPC from application's perspective",
+          API_LATENCY_PER_CALL,
+          AGGREGATION_WITH_BYTES_HISTOGRAM,
```
Comment (sanjaypujare) [inline census/src/main/java/io/grpc/census/internal/ObservabilityCensusConstants.java:53]: Why is this a `_WITH_BYTES_HISTOGRAM` ? The measure reported is `API_LATENCY_PER_CALL` and no bytes are reported. Wouldn't this be similar to `RpcViewConstants.RPC_CLIENT_ROUNDTRIP_LATENCY_VIEW` i.e. an aggregation over RPC millis?

Comment (sanjaypujare) [reply census/src/main/java/io/grpc/census/internal/ObservabilityCensusConstants.java:53]: Also if you had a test (similar to `CensusModulesTest.newTagsPopulateOldViews`) to test this new view you would have noticed the wrong type. Is it possible to add a test for the view?

File: census/src/test/java/io/grpc/census/CensusModulesTest.java [hunk 3]
```diff
@@ -716,13 +717,14 @@ private void assertRealTimeMetric(
     assertEquals(expectedValue, record.getMetricAsLongOrFail(measure));
   }
 
-  private void assertZeroRetryRecorded() {
+  private void assertPerCallMetrics(double expectedLatencyValue) {
```
Comment (sanjaypujare) [inline census/src/test/java/io/grpc/census/CensusModulesTest.java:720]: I see this is used to test the new metric but I don't see any of the retry tests (e.g. `recordRetryStats` on line 515) testing the new metric (total call latency which should be the sum of all retries latencies). May be you can update the `recordRetryStats` test?

**Review by sanjaypujare — CHANGES_REQUESTED** [review]
Found a potential issue and requesting some tests/changes

File: census/src/main/java/io/grpc/census/internal/ObservabilityCensusConstants.java (hunk 1 above)
Comment (DNVindhya) [reply census/src/main/java/io/grpc/census/internal/ObservabilityCensusConstants.java:51]: Done

File: census/src/main/java/io/grpc/census/internal/ObservabilityCensusConstants.java (hunk 2 above)
Comment (DNVindhya) [reply census/src/main/java/io/grpc/census/internal/ObservabilityCensusConstants.java]: You are right, updated measure to use `AGGREGATION_WITH_MILLIS_HISTOGRAM`. 
And added unit test for  `GRPC_CLIENT_API_LATENCY_VIEW`.

File: census/src/test/java/io/grpc/census/CensusModulesTest.java (hunk 3 above)
Comment (DNVindhya) [reply census/src/test/java/io/grpc/census/CensusModulesTest.java:725]: Updated

**Review by sanjaypujare — APPROVED** [review]
LGTM

**Pull request merged**